tests/unit/auth/v4/headerAuthCheck.js

```diff
@@ -36,6 +36,35 @@ const request = {
 };
 
 describe('v4 headerAuthCheck', () => {
+    it('should succeed if missing token', () => {
+        const clock = lolex.install(1454962445000);
+        const res = headerAuthCheck(request, log);
+        clock.uninstall();
+        assert.strictEqual(res.err, null);
+    });
+
+    it('shoud fail with an invalid token', () => {
+        const clock = lolex.install(1454962445000);
+        const alteredRequest = createAlteredRequest({
+            'x-amz-security-token': 'atoken' }, 'headers', request, headers);
+        const res = headerAuthCheck(alteredRequest, log);
+        clock.uninstall();
+        assert.notStrictEqual(res.err, null);
+        assert.strictEqual(res.err.InvalidToken, true);
+    });
+
+    it('shoud succeed on valid token', () => {
+        const clock = lolex.install(1454962445000);
+        const alteredRequest = createAlteredRequest(
+            { 'x-amz-security-token': 'a'.repeat(128) },
+            'headers', request, headers);
+        const res = headerAuthCheck(alteredRequest, log);
+        clock.uninstall();
+        if (res.err) {
```

Comment:
same ^^ if i test it here i have to throw in the next line, instead of the current assert